### PR TITLE
fix: add public_network_access_enabled to adapt azurerm 4.x + provider

### PR DESCRIPTION
**Pull request by jcframil**
fix: add public_network_access_enabled to adapt azurerm 4.x + provider

### CODE REVIEW

File: modules/azure-flexible-server-postgresql/flexible_server.tf [hunk 1]
```diff
@@ -11,6 +11,7 @@ resource "azurerm_postgresql_flexible_server" "postgresql_flexible_server" {
   version                           = local.data.server.version
   delegated_subnet_id               = local.subnet_id
   private_dns_zone_id               = data.azurerm_private_dns_zone.private_dns_zone.id
+  public_network_access_enabled     = local.data.public_network_access_enabled
```
Comment (alambike) [inline modules/azure-flexible-server-postgresql/flexible_server.tf:14]: And what is the default value of this setting :thinking:? I mean, if the input yaml doesn't define this attribute it should be false right?

Comment (jcframil) [reply modules/azure-flexible-server-postgresql/flexible_server.tf]: You are right, it should be `false` by default. I added it in commit [11c7e4d](https://github.com/prefapp/tfm/pull/143/commits/11c7e4db236719f24aa52029ff37bd4edad3b4fa)

```hcl
 public_network_access_enabled     = lookup(local.data.server, "public_network_access_enabled", false)
 ```

**Review by alambike — APPROVED** [review]

**Review by tonirvega — COMMENTED** [review]
I would add a `versions.tf` file. Besides that, if you need to declare outputs, I would declare it in `outputs.tf`

As terraform docs explain:  https://www.terraform-best-practices.com/code-structure#getting-started-with-the-structuring-of-terraform-configurations

**Comment by jcframil** [comment]
> I would add a `versions.tf` file. Besides that, if you need to declare outputs, I would declare it in `outputs.tf`
> 
> As terraform docs explain: https://www.terraform-best-practices.com/code-structure#getting-started-with-the-structuring-of-terraform-configurations

I'm not using outputs here, but I will define a versions.tf as suggests

**Pull request merged**